### PR TITLE
SelectBox: Set "aria-haspopup" and "aria-autocomplete" when fieldTemplate is rendered (T1230971)

### PR DESCRIPTION
**Pull request by jdvictoria**

### CODE REVIEW

**Comment by jdvictoria** [comment]
[SelectBox - fieldTemplate input doesn't contain the "aria-haspopup" and "aria-autocomplete" on the first render [T1230971]](https://trello.com/c/fdt5kkGl/8823-selectbox-fieldtemplate-input-doesnt-contain-the-aria-haspopup-and-aria-autocomplete-on-the-first-render-t1230971)

File: packages/devextreme/js/ui/drop_down_editor/ui.drop_down_editor.js [hunk 1]
```diff
@@ -411,6 +411,7 @@ const DropDownEditor = TextBox.inherit({
         this._refreshValueChangeEvent();
         this._refreshEvents();
         this._refreshEmptinessEvent();
+        this._setDefaultAria();
```
Comment (EugeniyKiyashko) [inline packages/devextreme/js/ui/drop_down_editor/ui.drop_down_editor.js:414]: We implemented the code for dropDownEditor, but the current test is designed for selectBox. While we can use the selectBox test to validate the customer's scenario, we should also create a test specifically for dropDownEditor

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/selectBox.tests.js [hunk 2]
```diff
@@ -6101,6 +6101,18 @@ if(devices.real().deviceType === 'desktop') {
...
         });
+
+        QUnit.test('aria-haspopup and aria-autocomplete attributes should exist when fieldTemplate is rendered (T1230696)', function(assert) {
+            helper.createWidget({
+                fieldTemplate: () => {
+                    return $('<div>').dxTextBox();
+                },
+            });
+
+            assert.ok(helper.$widget.find(`.${TEXTEDITOR_INPUT_CLASS}`).attr('aria-haspopup'), true);
```
Comment (EugeniyKiyashko) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/selectBox.tests.js]: These assertions and comments lack informativeness. What values are stored in these attributes? Can we use the strictEqual function here and provide corresponding comments for each attribute?

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/dropDownEditor.tests.js [hunk 3]
```diff
@@ -2097,6 +2097,19 @@ QUnit.module('aria accessibility', () => {
         assert.equal($input.attr('aria-expanded'), 'false', 'aria-expanded property on closed');
     });
 
+    QUnit.test('aria-haspopup and aria-autocomplete attributes should exist when fieldTemplate is rendered (T1230696)', function(assert) {
```
Comment (EugeniyKiyashko) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/dropDownEditor.tests.js]: You confused the ticket IDs with another bug. It should be (T1230971)

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/dropDownEditor.tests.js [hunk 4]
```diff
@@ -2097,6 +2097,19 @@ QUnit.module('aria accessibility', () => {
...
+        const $dropDownEditor = $('#dropDownEditorLazy').dxDropDownEditor({
+            fieldTemplate() {
+                return $('<div>').dxTextBox();
+            }
+        });
+        const $input = $dropDownEditor.find(`.${TEXT_EDITOR_INPUT_CLASS}`);
+
+        assert.strictEqual($input.attr('aria-haspopup'), 'true', 'fieldTemplate should have aria-haspopup attribute set to true');
```
Comment (EugeniyKiyashko) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/dropDownEditor.tests.js]: Let's check that after rerender of field these attributes are presented on input element.

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/dropDownEditor.tests.js [hunk 5]
```diff
@@ -2137,19 +2137,32 @@ QUnit.module('aria accessibility', () => {
         }).dxValidator({
             validationRules: [ { type: 'required' } ]
         });
+        const $input = $dropDownEditor.find(`.${TEXT_EDITOR_INPUT_CLASS}`);
```
Comment (EugeniyKiyashko) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/dropDownEditor.tests.js]: Will it the same input after rerender of the field? I suppose that no. We need to get new link to the element after field was rerendered.

**Review by EugeniyKiyashko — APPROVED** [review]

**Review by AlexanderMoiseev — APPROVED** [review]

**Pull request merged**